### PR TITLE
Tabs `from_payload` Update

### PR DESCRIPTION
**Pull request by lougeniaC64**
Not a whole lot to see here, just a persnickety PR. I hated the `match` statement I created to populate the `icon` field so I refactored `parse_string_from_json` to handle optional string values instead.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/tabs/src/sync/record.rs [hunk 1]
```diff
@@ -122,14 +122,25 @@ fn parse_last_used(last_used_val: &JsonValue) -> Result<u64> {
...
-        .ok_or_else(|| serde_json::Error::custom(format!("missing field `{}`", field_name)))?
-        .as_str()
-        .ok_or_else(|| {
-            serde_json::Error::custom(format!("invalid `{}`, expected string", field_name))
-        })?
-        .to_string())
+fn parse_string_from_json(
+    field_name: &str,
+    val: Option<&JsonValue>,
+    can_be_empty: bool,
```
Comment (rfk) [inline components/tabs/src/sync/record.rs:128]: Naming consistency nit: "empty" -> "missing" for consistency with the error message that you throw when this value is `false`.

**Review by rfk — APPROVED** [review]
Makes sense to me. I'm not a big fan of boolean flag arguments like this but I can't argue with the resulting code cleanup :-)

**Comment by lougeniaC64** [comment]
> Makes sense to me. I'm not a big fan of boolean flag arguments like this but I can't argue with the resulting code cleanup :-)



> Makes sense to me. I'm not a big fan of boolean flag arguments like this but I can't argue with the resulting code cleanup :-)

That's fair. I created a separate function for the optional string and got rid of the boolean flag.

**Comment by lougeniaC64** [comment]
Closing as these `from_record` changes are being reverted in #4337.

**Pull request closed without merging**